### DISCUSSION TITLE
Use `code-server` command locally to open folder remotely

### DISCUSSION THREAD

**Discussion by bchandos** [coder/code-server#4999] [Q&A]
Hello! My setup is a docker container running on my local machine (Fedora 35), with the PWA installed locally via the Flatpak version of Chromium.

What I'm trying to accomplish is being able to type `code-server .` in a folder on my local machine and have it open an instance of the PWA starting in that folder, much like `code .` would with VSCode.

The remote folders are mapped to local folders in `docker-compose.yaml` using `volumes:` so it would be easy to make that translation and pass the correct folder name from code-server's perspective. What I can't figure out is how to run the flatpak command and pass the `folder` query parameter to Chromium.

The command that the `.desktop` file executes:
```
flatpak 'run' '--command=/app/bin/chromium' 'org.chromium.Chromium' '--profile-directory=Default' '--app-id=<app-id>'
```

I tried `--extra-search-query-params` but that doesn't seem to work (I think it's a debugging tool for adding query parameters only to search queries).

I can run it with the URL directly, like:
```
flatpak 'run' '--command=/app/bin/chromium' 'org.chromium.Chromium' '--profile-directory=Default' 'http://localhost:8080?folder=/home/coder/project/my_app'
```
... but that opens in the Chromium browser as a tab, instead of launching the installed PWA version.

Any ideas? Thanks!

**Comment by lg-nineoneninethreenine** [comment]
I don't know if you can directly access PWAs during the command line, but if you can, don't question me! I use Windows with just launching my browser and visiting the server from my Linux machines.

**Comment by jsjoeio** [comment]
Have you tried something [like this](https://stackoverflow.com/a/67552053/3015595)?

**Reply by bchandos** [reply]
Thanks for the reply. Unless I am misunderstanding, that solution appears more geared for passing multiple app IDs to Chromium. I think I need to pass the `folder` query parameter to a single, unchanging, app ID, and am unsure how to do so via the command-line.

I guess this is ultimately a Chromium question, but I thought it might have come up here before.

**Comment by bchandos** [comment]
One possible solution is to pass the full URL with `folder` parameter to Chromium using the `--app` switch. This will open the code editor as an app. However, it will appear in the application manager as an instance of Chromium, instead of as the installed `code-server` PWA. I'm not sure yet if this would be a "deal-breaker". I suspect it would cause task and window switching to be more confusing: instead of all `code-server` PWA windows being under a single, separate, application, they will become entangled with any other Chromium windows that are open.

Still, it's something.

```
flatpak 'run' '--command=/app/bin/chromium' 'org.chromium.Chromium' '--profile-directory=Default' '--app=http://localhost:8080?folder=/home/coder/project/my_app'
```

**Reply by jsjoeio** [reply]
Hmm 🤔 I've had similar issues on my PC because my window manager thinks they're all the same application.

Maybe @code-asher has a suggestion?

**Reply by code-asher** [reply]
Unfortunately I have no experience here :disappointed:
